sklearn/preprocessing/tests/test_data.py

```diff
@@ -1500,13 +1506,37 @@ def test_one_hot_encoder_sparse():
     # test exception on wrong init param
     assert_raises(TypeError, OneHotEncoder(n_values=np.int).fit, X)
 
+
+def test_one_hot_encoder_error_on_negative():
+    # Negative numerical values in inputs should raise an exception
```

Comment:
Maybe I should say that the explicit check in `fit` is only when `values="auto"` and only on integer features. That mode is still assuming that the feature is a 0-based ordinal. I'm not sure what better way there is to not break the way the OHE used to work.